web/client/components/share/SharePanel.jsx

```diff
@@ -106,8 +106,8 @@ class SharePanel extends React.Component {
         modal: false,
         draggable: true,
         onClose: () => {},
-        shareUrlRegex: "(h[^#]*)#\\/viewer\\/([^\\/]*)\\/([A-Za-z0-9]*)",
-        shareUrlReplaceString: "$1embedded.html#/$3",
+        shareUrlRegex: "(h[^#]*)#\/viewer\/([^\/]*\/[A-Za-z0-9]*|[A-Za-z0-9]*)",
```

Comment:
I approved but I have a doubt: are these changes really needed, in particular `\/[A-Za-z0-9]*|[A-Za-z0-9]*`, shouldn't be enough to remove `\\/([^\\/]*)` from the `shareUrlRegex` current on master?
Could you try with the follow one instead for shareUrlRegex prop?

`(h[^#]*)#\\/viewer\\/([A-Za-z0-9]*)`

Reply:
I was thinking the map URL may include 2 terms after viewer like: 
* http://hostname/#/viewer/shared/{map id}.
 
I have tried your suggestion and it works well